server/room.ts

```diff
@@ -134,14 +142,39 @@ export class Room {
       });
       socket.on('CMD:getRoomState', (data) => this.getRoomState(socket));
       socket.on('CMD:setRoomState', (data) => this.setRoomState(socket, data));
-      socket.on('CMD:setRoomOwner', (data) => this.setRoomOwner(socket, data));
+      socket.on('CMD:setRoomOwner', (data) =>
+        this.setRoomOwner(socket, io.of(roomId), data)
+      );
       socket.on('signal', (data) => this.sendSignal(socket, data));
       socket.on('signalSS', (data) => this.signalSS(socket, data));
 
       socket.on('disconnect', () => this.disconnectUser(socket));
     });
   }
 
+  setRoomCapacity = (isOwnerSub: boolean) => {
+    this.roomCapacity = isOwnerSub
+      ? config.ROOM_CAPACITY_SUB
+      : config.ROOM_CAPACITY;
+  };
+
+  checkSub = async () => {
```

Comment:
I'd actually be fine with rooms retaining their larger capacity even after the owner's sub expires (they just can't make any new large rooms). This also aligns it with the current behavior of vanity names. If we decide we want to clean them up later we can do both those things in a cleanup function.

That might skip the need for checking the sub on each room when it's rebuilt from redis (which may add to startup time as we're hitting the database thousands of times)

Reply:
I thought about adding a new column "isSubRoom" to the table for this. Does that sound fine?

Reply:
Yeah, I think that'd be fine. We just want to avoid the performance penalty of having to check sub state for thousands of rooms on boot.

Reply:
I will try to solve the  "upgrade" problem after you have approved the current changes

Reply:
this looks fine to me. However, the upgrade problem is going to require one of the following:
* Keep a copy of subscribers in our local db
* have stripe notify us somehow when a new subscription happens (but this notification might be lost so we need a backup to make this consistent)

Reply:
How about stripe webhook notifications?

Reply:
> If Stripe does not receive a 2xx HTTP status code, the notification attempt is repeated.

Reply:
Or maybe we could just upgrade the room(s) when the subscriber logs in and ask people to relog into their accounts if they want their rooms to be upgraded.

Reply:
Actually no, forget that...